libpromises/signals.c

```diff
@@ -155,6 +155,11 @@ void HandleSignalsForAgent(int signum)
     case SIGINT:
         /* TODO don't exit from the signal handler, just set a flag. Reason is
          * that all the cleanup() hooks we register are not reentrant. */
+
+        /* Send corresponding signal to process group so that custom promise
+         * modules and backgrounded processes spawned by the commands promise
+         * type gets terminated as well. */
+        killpg(0, signum);
```

Comment:
AFAICT, this sends the signal to the sending process as well because it's part of the same process group. Potentially causing all sorts of weird things like what the `TODO` above mentions.

Reply:
It seemed to work except for windows build. But yes, I've read that the signal handler may be called again after exiting. And that is weird indeed. I'm trying a new solution where I'm using process select to find all processes that is not the agent it self, but has the same PGID, signalling their PID's individually using `GracefulTerminate`.

Reply:
Sounds promising!